cmd/server-test/error_analyzer_test.go

```diff
@@ -72,9 +89,18 @@ func (suite *ErrorAnalyzerIntegrationSuite) TearDownTest() {
 func (suite *ErrorAnalyzerIntegrationSuite) TestAnalyzerErr() {
 	suite.sendEvent(successJSON)
 
-	suite.GrepTrue(suite.r, `msg="analysis failed" analyzer=Dummy app=lookoutd error="rpc error: code = Unknown desc = review error"`)
+	suite.GrepTrue(suite.r, suite.errMessage)
 }
 
 func TestErrorAnalyzerIntegrationSuite(t *testing.T) {
-	suite.Run(t, new(ErrorAnalyzerIntegrationSuite))
+	suite.Run(t, &ErrorAnalyzerIntegrationSuite{
```

Comment:
Usually `Test...Suite` is just a wrapper for one `stretchr/testify` suite. With these two `suite.Run` inside the same test we get:
```
go test -v -parallel 1 -tags='integration' github.com/src-d/lookout/cmd/server-test -run TestErrorAnalyzerIntegrationSuite/TestAnalyzerErr

=== RUN   TestErrorAnalyzerIntegrationSuite
=== RUN   TestErrorAnalyzerIntegrationSuite/TestAnalyzerErr
=== RUN   TestErrorAnalyzerIntegrationSuite/TestAnalyzerErr#01
--- PASS: TestErrorAnalyzerIntegrationSuite (15.60s)
    --- PASS: TestErrorAnalyzerIntegrationSuite/TestAnalyzerErr (8.33s)
    --- PASS: TestErrorAnalyzerIntegrationSuite/TestAnalyzerErr#01 (7.26s)
```

If one of the suites fails, it's hard to know which one is it. Also, it's hard to run only one of them, since they are bundled togueter.

Since you made a different mock analyzer for the timeout error, you could have two `Test...Suite`, probably in a different file, e.g. `timeout_analyzer_test.go`. `startAnalyzer` could be moved to `common_test.go`, maybe renaming it to `startMockAnalyzer` to avoid confusion.

Reply:
Addressed [here](https://github.com/src-d/lookout/pull/442/commits/091605f9521a3a311766a8f32aa069a13108d257)